cfme/tests/containers/test_autotagging_providers.py

```diff
@@ -0,0 +1,198 @@
+from collections import namedtuple
+import fauxfactory
+import pytest
+from random import choice
+from traceback import format_exc
+
+from cfme.configure.configuration.region_settings import MapTags
+from cfme.containers.provider import ContainersProvider
+from cfme.containers.node import Node
+from cfme.containers.replicator import Replicator
+from cfme.containers.image import Image
+from cfme.containers.pod import Pod
+from cfme.containers.project import Project
+from cfme.containers.route import Route
+
+from utils import testgen
+from utils.version import current_version
+from utils.wait import wait_for
+from utils.log import logger
+
+
+pytestmark = [
+    pytest.mark.uncollectif(lambda: current_version() < "5.8"),
+    pytest.mark.usefixtures('setup_provider_modscope'),
+    pytest.mark.tier(1)]
+pytest_generate_tests = testgen.generate([ContainersProvider], scope='module')
+
+TEST_OBJECTS = (Pod, Route, Node, Replicator, Project)
+
+values = ['{}{}'.format('value_', fauxfactory.gen_alpha(5).lower()),
+          '{}{}'.format('value_', fauxfactory.gen_alphanumeric(5).lower()),
+          '{}{}{}'.format('value_', fauxfactory.gen_alpha(3).lower(),
+                          fauxfactory.gen_alphanumeric(3))]
+
+
+def check_smart_management_in_ui(instance, expected_value):
+    return instance.summary.smart_management.my_company_tags[0].value == str(expected_value)
+
+
+def set_label(test_obj, instance, label_name, value):
+    try:
+        status_code, json_content = test_obj.set_label(instance, label_name, value)
+        # temp debugging, remove later
+        print('setting label status code is: ' + str(status_code))
+    except:
+        status_code, json_content = None, format_exc()
+    return status_code, json_content
+
+
+def remove_labels(data_collection):
+    for instance, company_tag, set_value, category, results_status, \
+            results_message, label_name in data_collection:
+                try:
+                    instance.remove_label(label_name)
+                except:
+                    logger.warning('Cannot remove label ({} = {}) for {} {}. '
+                                   '(failed to add it previously)'
+                                   .format(label_name, set_value,
+                                           instance.__class__.__name__, instance.name))
+
+
+def add_tag(instance, label_name, category):
```

Comment:
**OBSERVATION**: A note here that we need to change MapTags, into using NavigatableMixin and not Navigatable